crates/distribution-types/src/file.rs

```diff
@@ -153,6 +151,24 @@ impl Display for FileLocation {
 #[archive_attr(derive(Debug))]
 pub struct UrlString(String);
 
+impl serde::Serialize for UrlString {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        String::serialize(&self.0, serializer)
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for UrlString {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        String::deserialize(deserializer).map(UrlString)
+    }
+}
+
```

Comment:
Are these `serde(transparent)`, or am i missing something?

Reply:
Yes thank you! https://github.com/astral-sh/uv/pull/6633